agent/agent.go

```diff
@@ -34,28 +33,28 @@ type Agent struct {
 	metadata           map[string]string
 	validator          validation.Strategy
 	httpClient         RequestDoer
+	dialInsecure       bool
```

Comment:
NIT: I'm not a fan of bool arguments to functions. `dialInsecure` could be an exported field, treated the same way as we treat `HTTPDisableResponseCompression`.

Reply:
I don't mind having non-essential options as exported struct fields, but this solution is excellent.